src/backend/distributed/commands/function.c

```diff
@@ -192,6 +198,45 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
 
+	DistObjectCacheEntry *cacheEntry =
+		LookupDistObjectCacheEntry(ProcedureRelationId, funcOid, InvalidOid);
+	if (cacheEntry != NULL && IsObjectDistributed(&functionAddress))
+	{
+		/*
+		 * If the colocationId, forceDelegation and distributionArgIndex fields of a
+		 * pg_dist_object entry of a distributed function are all set to zero, it means
+		 * that function is either automatically distributed by ddl propagation, without
+		 * calling create_distributed_function. Or, it could be distributed via
+		 * create_distributed_function, but with no parameters.
+		 *
+		 * For these cases, calling create_distributed_function for that function,
+		 * without parameters would be idempotent. Hence we can simply early return here,
+		 * by providing a notice message to the user.
+		 */
+		bool idempotentDistribution = /* are pg_dist_object fields set to zero? */
```

Comment:
Let's create 2 different variable
i) For checking whether function related columns on pg_dist_object are 0.
ii) For checking whether the function is called without parameters.